crates/runtime/src/actor/component/http.rs

```diff
@@ -33,23 +35,62 @@ pub mod incoming_http_bindings {
 }
 
 impl WasiHttpView for Ctx {
-    fn table(&mut self) -> &mut Table {
-        &mut self.table
-    }
-
     fn ctx(&mut self) -> &mut WasiHttpCtx {
         &mut self.http
     }
 
+    fn table(&mut self) -> &mut Table {
+        &mut self.table
+    }
+
     fn send_request(
         &mut self,
-        _request: OutgoingRequest,
+        OutgoingRequest {
+            use_tls,
+            authority,
+            request,
+            connect_timeout,
+            first_byte_timeout,
+            between_bytes_timeout,
+        }: OutgoingRequest,
     ) -> wasmtime::Result<Resource<HostFutureIncomingResponse>>
     where
         Self: Sized,
     {
-        // TODO: Implement https://github.com/wasmCloud/wasmCloud/issues/680
-        bail!("outgoing HTTP not supported yet")
+        let request = request.map(|body| -> Box<dyn AsyncRead + Send + Sync + Unpin> {
+            Box::new(BodyAsyncRead::new(body))
+        });
+        let handler = self.handler.clone();
+        let res = HostFutureIncomingResponse::new(preview2::spawn(async move {
+            match OutgoingHttp::handle(
+                &handler,
+                OutgoingHttpRequest {
+                    use_tls,
+                    authority,
+                    request,
+                    connect_timeout,
+                    first_byte_timeout,
+                    between_bytes_timeout,
+                },
+            )
+            .await
+            {
+                Ok(resp) => {
+                    let resp = resp.map(|body| BoxBody::new(AsyncReadBody::new(body, 1024)));
+                    Ok(IncomingResponseInternal {
+                        resp,
+                        worker: preview2::spawn(async { Ok(()) }),
+                        between_bytes_timeout,
+                    })
+                }
+                Err(e) => Err(e),
```

Comment:
If we don't need to do anything with the error case, how about `map` instead of `match`?

Reply:
Sounds good, but I will not be addressing that here to avoid extending the review cycle, since this PR is blocking #745 and potentially may cause conflicts with #807